pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,254 @@
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	o.orgConfig = config
+	// required for the way existing getClient function is designed
+	o.orgConfig.AccountRegion = "us-west-2"
+
+	// While doing configuration invalidate the map so we dont keep using old one.
+	o.orgListAccountMap = make(map[string]bool)
+	o.retries = orgAccountRetries
+
+	t, err := time.ParseDuration(config.AccountListTTL)
+	if err != nil {
+		return status.Errorf(codes.Internal, "issue while parsing ttl for organization, while configuring orgnization validation: %v", err)
+	}
+
+	o.orgAccountListCacheTTL = t
+
+	return nil
+}
+
+func (o *orgValidator) setLogger(log hclog.Logger) {
+	o.log = log
+}
+
+// This verification method checks if the Account ID attached on the node is part of the organisation.
+// If it part of the organisation then validation should be succesfull if not attestation should fail, on enabling this verification method.
+// This could be alternative for not explictly maintaing allowed list of account ids.
+// Method pulls the list of accounts from organization and caches it for certain time, cache time can be configured.
+func (o *orgValidator) IsMemberAccount(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDofNode string) (bool, error) {
+
+	reValidatedcache, err := o.validateCache(ctx, orgClient)
+	if err != nil {
+		return false, err
+	}
+
+	accountIsmemberofOrg, err := o.lookupCache(ctx, orgClient, accoundIDofNode, reValidatedcache)
+	if err != nil {
+		return false, err
+	}
+
+	return accountIsmemberofOrg, nil
+}
+
+// Validate cache and refresh if its stale
+func (o *orgValidator) validateCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (bool, error) {
+
+	isStale, err := o.checkIfOrgAccountListIsStale(ctx)
+	if err != nil {
+		return isStale, err
+	}
+
+	// refresh the account map
+	if isStale {
+		_, err = o.reloadAccountList(ctx, orgClient, false)
+		if err != nil {
+			return isStale, err
+		}
+	}
+	return isStale, nil
+}
+
+func (o *orgValidator) lookupCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDofNode string, reValidatedcache bool) (bool, error) {
+
+	o.mutex.RLock()
+	orgAccountList := o.orgListAccountMap
+	o.mutex.RUnlock()
+
+	_, accoutIsmemberOfOrg := orgAccountList[accoundIDofNode]
+
+	// Retry if it doesn't exist in cache and cache was not revalidated
+	if !accoutIsmemberOfOrg && !reValidatedcache {
+		orgAccountList, err := o.refreshCache(ctx, orgClient)
+		if err != nil {
+			return false, err
+		}
+		_, accoutIsmemberOfOrg = orgAccountList[accoundIDofNode]
+	}
+
+	return accoutIsmemberOfOrg, nil
+}
+
+// If cache miss happens, refresh list with new cache and check if element exist
+func (o *orgValidator) refreshCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (map[string]bool, error) {
+	remTries := o.getRetries()
+
+	orgAccountList := make(map[string]bool)
+	if remTries <= 0 {
+		return orgAccountList, nil
```

Comment:
why is it required to return an empty list when there are no retries? 
when is retry reset?  can no we get into a situation where we start to return empty maps always?